#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_missing-header.dpatch by Ming Hua <minghua-guest@users.alioth.debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Adding missing header inclusions to fix FTBFS with GCC 4.3.

@DPATCH@
diff -urNad scim-hangul-0.3.1~/src/scim_hangul_imengine.cpp scim-hangul-0.3.1/src/scim_hangul_imengine.cpp
--- scim-hangul-0.3.1~/src/scim_hangul_imengine.cpp	2007-01-06 08:00:57.000000000 -0600
+++ scim-hangul-0.3.1/src/scim_hangul_imengine.cpp	2007-10-02 11:29:06.000000000 -0500
@@ -28,6 +28,7 @@
 #define Uses_SCIM_IMENGINE
 #define Uses_SCIM_LOOKUP_TABLE
 #define Uses_SCIM_CONFIG_BASE
+#define Uses_C_STRING
 
 #ifdef HAVE_CONFIG_H
   #include <config.h>
diff -urNad scim-hangul-0.3.1~/src/scim_hangul_imengine_setup.cpp scim-hangul-0.3.1/src/scim_hangul_imengine_setup.cpp
--- scim-hangul-0.3.1~/src/scim_hangul_imengine_setup.cpp	2006-10-23 07:42:47.000000000 -0500
+++ scim-hangul-0.3.1/src/scim_hangul_imengine_setup.cpp	2007-10-02 11:29:06.000000000 -0500
@@ -27,6 +27,7 @@
  */
 
 #define Uses_SCIM_CONFIG_BASE
+#define Uses_C_STRING
 
 #include <gtk/gtk.h>
 
